orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/MonitoredDeployStrategy.groovy

```diff
@@ -404,8 +404,13 @@ class MonitoredDeployStrategy implements Strategy {
       internalStageData.cloudProvider = baseContext.cloudProvider
       internalStageData.region = baseContext.region
       internalStageData.oldServerGroup = source?.serverGroupName
-      internalStageData.newServerGroup = stageData.deployServerGroups[baseContext.region].first()
-      internalStageData.parameters = stageData.deploymentMonitor.parameters
+      //TODO (gyardeni): check if it is true here?
+      //internalStageData.failure = true;
+
+      if (stageData.deployServerGroups && !stageData.deployServerGroups.isEmpty()) {
```

Comment:
fyi, i refactored this a bit here: https://github.com/spinnaker/orca/pull/3222/files#diff-cc03fa80e1a96b9c0443a8fe1ef22a6fR133
so this should no longer be needed